---
title: "EdgeR and MA plot of nuclei RNA-seq"
author: "Evangelyn Sim"
date: "2/19/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```




## Introduction

In the GEO submission 1 processed files were uploaded. 

1. mrna_fulllen_pe_strrev.mx.fix_filt

They have been uploaded in the /output folder and will be used below to generate different figures.



### Used libraries and functions

```{r}
library(edgeR)
library(limma)
library(dplyr)

```


### Read files 
```{r}

PRIOR = 20
FDR = 0.05

rm1 <- read.delim("/group/card2/Evangelyn_Sim/Collaboration_Kev_UoM/Sequencing_ATAC_RNA/Github/2021_UoM_Yap_shRNA_nuclei_RNAseq_ATACseq/output/mrna_fulllen_pe_strrev.mx.fix_filt", header = TRUE, row.names = 1)
matrix = rm1[,c(1:6)]

phenoN = data.frame(colnames(matrix))
colnames(phenoN) = "sampleN"

phenoN$sample = gsub(".[0-9]", "", phenoN$sampleN)

pheno = data.frame(phenoN[,c(2)])
colnames(pheno) =  "sample"


write.table(pheno, file="/group/card2/Evangelyn_Sim/Collaboration_Kev_UoM/Sequencing_ATAC_RNA/Github/2021_UoM_Yap_shRNA_nuclei_RNAseq_ATACseq/output/pheno.matrix.txt", sep="\t", quote = F, row.names = F)
pheno = read.delim(file="/group/card2/Evangelyn_Sim/Collaboration_Kev_UoM/Sequencing_ATAC_RNA/Github/2021_UoM_Yap_shRNA_nuclei_RNAseq_ATACseq/output/pheno.matrix.txt")

########################################################################################################

attach(pheno)
design = model.matrix(as.formula("~ 0  + sample"))
detach(pheno)
design

D = DGEList(counts=matrix)
D = calcNormFactors(D)
D = estimateGLMCommonDisp(D, design)
D = estimateGLMTagwiseDisp(D, design, prior.df = PRIOR)
fit = glmFit(D, design, prior.count = PRIOR)

Contrast = makeContrasts(rna_LacZvsYap = sampleYap - sampleLacz,
                         levels=design)

res = list()
contrast.name = colnames(Contrast)

for(i in 1:length(contrast.name)){
  lrt = glmLRT(fit, contrast = Contrast[,i])   
  
  results = lrt$table
  disp = lrt$dispersion
  fitted.vals = lrt$fitted.values
  coefficients = lrt$coefficients
  
  results$adj.p.value = p.adjust(p = results$PValue, method = "fdr" )
  table(row.names(results) == row.names(fitted.vals))
  
  Name = row.names(results)
  res0 = cbind(Name, results, disp, fitted.vals, coefficients)
  res[[i]] = res0[order(res0$adj.p.value),]
  write.table(res[[i]], file= paste0("/group/card2/Evangelyn_Sim/Collaboration_Kev_UoM/Sequencing_ATAC_RNA/Github/2021_UoM_Yap_shRNA_nuclei_RNAseq_ATACseq/output/edgeR_", contrast.name[i] ,".xls"), quote=F, sep="\t", col.names = T, row.names = F)
  
  
  res[[i]]= mutate(res[[i]], cs= ifelse(res[[i]]$adj.p.value <= 0.05 , "red", "black"))
  
  mxFDR = res[[i]][res[[i]]$adj.p.value <= FDR,]
  mxFDR_Up = mxFDR[mxFDR$logFC>0,]
  mxFDR_Dn = mxFDR[mxFDR$logFC<0,]
  
  res[[i]]= mutate(res[[i]], FDR= nrow(mxFDR))
  res[[i]]= mutate(res[[i]], FDRup= nrow(mxFDR_Up))
  res[[i]]= mutate(res[[i]], FDRdn= nrow(mxFDR_Dn))

}

```

```{r, fig.height=5, fig.width= 5}

par(mfrow=c(1,1))

for(i in 1:length(contrast.name)){

  plot(res[[i]]$logFC, -log10(res[[i]]$PValue), pch=20, cex=0.8, col=res[[i]]$cs, 
       xlab = "LogFC", ylab = "-log10(PValue)",
       main = paste0(contrast.name[i], 
                      "\nFDR<=0.05, N=", res[[i]][1,ncol(res[[i]])-2], 
                      "\nUp=",res[[i]][1,ncol(res[[i]])-1],", Dn=",res[[i]][1,ncol(res[[i]])]))
  grid(nx = NULL, ny = NULL, col = "blue", lty = "dotted")
  
  
}

```

